server/fishtest/actiondb.py

```diff
@@ -8,17 +8,33 @@ def __init__(self, db):
         self.db = db
         self.actions = self.db["actions"]
 
-    def get_actions(self, max_num, action=None, username=None, before=None):
+    def get_actions(
+        self,
+        username = None,
+        action = None,
+        limit = 0,
+        skip = 0,
+        utc_before = None,
+        max_actions = None,
+    ):
         q = {}
         if action:
             q["action"] = action
         else:
             q["action"] = {"$nin": ["update_stats", "dead_task"]}
         if username:
             q["username"] = username
-        if before:
-            q["time"] = {"$lte": before}
-        return self.actions.find(q, sort=[("_id", DESCENDING)], limit=max_num)
+        if utc_before:
+            q["time"] = {"$lte": utc_before}
+
+        count = self.actions.count_documents(q)
+        if max_actions:
+            limit = min(limit, max_actions)
```

Comment:
```python
            limit = min(limit, max_actions - skip)
```
In this way we have the right last page when `max_actions` is bigger than and not multiple of `page_size`:
https://dfts-0.pigazzini.it/actions?user=ChessDBCN&action=failed_task&max_actions=53&before=1664312574.521